### PR TITLE
Feature/turret talonfx

### PR DESCRIPTION
**Pull request by NoamRosenberg08**

### CODE REVIEW

File: src/main/java/frc/robot/turret/talonfx/TalonFXTurret.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    private final StatusSignal<Double> velocitySignal;
+    private final StatusSignal<Double> voltageSignal;
+    private final StatusSignal<Double> currentSignal;
+
+    public TalonFXTurret(CTREDeviceID turretID, TalonFXConfiguration motorConfiguration) {
+        this.motor = new TalonFXWrapper(turretID);
+        this.motor.applyConfiguration(motorConfiguration);
+
+        BaseStatusSignal.setUpdateFrequencyForAll(GlobalConstants.DEFAULT_CAN_FREQUENCY);
```
Comment (YHerm) [inline src/main/java/frc/robot/turret/talonfx/TalonFXTurret.java]:
```suggestion
        BaseStatusSignal.setUpdateFrequencyForAll(GlobalConstants.DEFAULT_CAN_FREQUENCY, signals...);
```

File: src/main/java/frc/robot/turret/talonfx/TalonFXTurret.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    @Override
+    public void stop() {
+        motor.stopMotor();
+    }
+
+    @Override
+    public void updateInputs(TurretInputsAutoLogged inputs) {
+        BaseStatusSignal.refreshAll(velocitySignal, voltageSignal, currentSignal);
```
Comment (YHerm) [inline src/main/java/frc/robot/turret/talonfx/TalonFXTurret.java:70]: add connected input that gets the status code of this refresh

File: src/main/java/frc/robot/turret/talonfx/TalonFXConstants.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package frc.robot.turret.talonfx;
+
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+
+public class TalonFXConstants {
```
Comment (YHerm) [inline src/main/java/frc/robot/turret/talonfx/TalonFXConstants.java]: TalonFXTurretConstants

File: src/main/java/frc/robot/turret/talonfx/TalonFXConstants.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+    public static final TalonFXConfiguration TURRET_MOTOR_CONFIGURATION = new TalonFXConfiguration();
```
Comment (YHerm) [inline src/main/java/frc/robot/turret/talonfx/TalonFXConstants.java]: MOTOR_CONFIGURATION

File: src/main/java/frc/robot/turret/talonfx/TalonFXTurret.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    private final StatusSignal<Double> currentSignal;
+    private final StatusSignal<Double> positionSignal;
+    private final StatusSignal<Double> accelerationSignal;
+
+    public TalonFXTurret(CTREDeviceID motorID, TalonFXConfiguration motorConfiguration) {
+        this.motor = new TalonFXWrapper(motorID);
+        motor.applyConfiguration(motorConfiguration);
+
+        this.positionDutyCycle = new PositionDutyCycle(0);
+        this.velocityDutyCycle = new VelocityDutyCycle(0);
```
Comment (YHerm) [inline src/main/java/frc/robot/turret/talonfx/TalonFXTurret.java:33]: what about enable/disable foc

**Review by YHerm — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/turret/talonfx/TalonFXTurret.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    private final TalonFXWrapper motor;
+
+    private final VelocityDutyCycle velocityDutyCycle;
+    private final PositionDutyCycle positionDutyCycle;
+
+    private final StatusSignal<Double> velocitySignal;
+    private final StatusSignal<Double> voltageSignal;
+    private final StatusSignal<Double> currentSignal;
+    private final StatusSignal<Double> positionSignal;
+    private final StatusSignal<Double> accelerationSignal;
```
Comment (YHerm) [inline src/main/java/frc/robot/turret/talonfx/TalonFXTurret.java:26]: keep it the same order everywhere:) 
(position, velocity, acceleration, current, voltage)

File: src/main/java/frc/robot/turret/talonfx/TalonFXTurret.java (hunk 2 above)
Comment (NoamRosenberg08) [reply src/main/java/frc/robot/turret/talonfx/TalonFXTurret.java:70]: for another branch :) i'll open it in a sec

**Pull request merged**